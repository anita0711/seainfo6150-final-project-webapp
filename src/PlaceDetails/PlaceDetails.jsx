import React from "react";
import { Link } from "react-router-dom";
import styles from "./PlaceDetails.module.css";

const PlaceDetails = (props) => {
  return (
    <div>
      <div className={styles.imageSection}>
        <img src={props.img1} alt={props.title} />
        <img src={props.img2} alt={props.title} />
        <img src={props.img3} alt={props.title} />
        <img src={props.img4} alt={props.title} />
        <img src={props.img5} alt={props.title} />
      </div>
      <p>
        <h3>Details:</h3>
        State: {props.state} <br />
        About: {props.about} <br />
        Description: {props.description} <br />
        Best for: {props.bestFor} <br />
        Things to do: {props.thingsToDo} <br />
        Best time to visit: {props.timeToVisit} <br />
        Dining: {props.dining} <br />
        Hotels: <a href={props.hotels}>See here</a> <br />
        Map: <a href={props.map}>See here</a> <br />
      </p>
    </div>
  );
};

export default PlaceDetails;
