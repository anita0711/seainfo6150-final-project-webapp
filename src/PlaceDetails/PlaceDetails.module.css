.imageSection {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  list-style: none;
  column-gap: 25px;
  row-gap: 25px;
  padding-left: 10px;
  justify-items: center;
  align-items: center;
}

.imageSection img {
  width: 80%;
  height: 200px;
  border-radius: 20px;
  margin: 0px;
  border: 5px solid #ccc;
  box-shadow: 0px 0px 20px black;
}

.imageSection img:hover {
  box-shadow: 10px 10px 20px black;
}

.detailsContainer {
  display: grid;
  grid-template-columns: 1fr 7fr;
  column-gap: 20px;
  margin-left: 20px;
}

.detailsTag {
  display: inline-block;
  background-color: rgb(209, 133, 83);
  color: white;
  border-radius: 10px;
  margin-bottom: 15px;
  margin-left: 10px;
  margin-top: 10px;
}

.details {
  margin-left: 20px;
}
