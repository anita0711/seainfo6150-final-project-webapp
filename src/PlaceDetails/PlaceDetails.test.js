import React from "react";
import { render } from "@testing-library/react";
import PlaceDetails from "./PlaceDetails.jsx";

it("renders the PlaceDetails component correctly", () => {
  const { container } = render(
    <PlaceDetails
      state="California"
      about="Ideal destination for adventurers, hikers and sun seekers"
      thingsToDo="Balboa Park, Coronado Beach, Mission and Pacific Beach"
      dining="Bobboi Natural Gelato, Ambrogio15, Parma Cucina Italiana"
    />
  );
  expect(container).toMatchSnapshot();
});
