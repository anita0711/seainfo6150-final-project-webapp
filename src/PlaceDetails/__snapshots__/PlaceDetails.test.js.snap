// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the PlaceDetails component correctly 1`] = `
<div>
  <div>
    <div
      class="imageSection"
    >
      <img />
      <img />
      <img />
      <img />
      <img />
    </div>
    <div
      class="detailsTag"
    >
      <h3>
        Details:
      </h3>
    </div>
    <div
      class="detailsContainer"
    >
      <p>
        State:
      </p>
      <p>
        California
         
      </p>
      <p>
        About:
      </p>
      <p>
        Ideal destination for adventurers, hikers and sun seekers
      </p>
      <p>
        Description:
      </p>
      <p />
      <p>
        Best for:
      </p>
      <p>
         
      </p>
      <p>
        Things to do:
      </p>
      <p>
        Balboa Park, Coronado Beach, Mission and Pacific Beach
      </p>
      <p>
        Best time to visit:
      </p>
      <p />
      <p>
        Dining:
      </p>
      <p>
        Bobboi Natural Gelato, Ambrogio15, Parma Cucina Italiana
      </p>
      <p>
        Hotels:
      </p>
      <p>
        <a>
          See here
        </a>
      </p>
      <div>
        Map:
      </div>
      <p>
        <a>
          See here
        </a>
      </p>
    </div>
  </div>
</div>
`;
